frameworks/MLNet/exec.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import math
+import os
+import pandas as pd
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from amlb.datautils import reorder_dataset
+from amlb.results import NoResultError, save_predictions
+from amlb.utils import dir_of, path_from_split, run_cmd, split_path, Timer
+import json
+from frameworks.shared.callee import call_run, result, save_metadata
+
+log = logging.getLogger(__name__)
+
+def run(dataset: Dataset, config: TaskConfig):
+    avaible_task_list = ['classification', 'regression']
+    if config.type not in avaible_task_list:
+        raise ValueError('{} is not supported.'.format(config.type))
+
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    DOTNET_INSTALL_DIR = os.path.join(dir_path, '.dotnet')
+    os.environ['MODELBUILDER_AUTOML'] = 'NNI'
+    os.environ['DOTNET_ROOT'] = DOTNET_INSTALL_DIR
+    mlnet = os.path.join(DOTNET_INSTALL_DIR, 'mlnet')
+    save_metadata(config)
+    name = config.name
+    temp_output_folder = os.path.join(config.output_dir,name)
+    if not os.path.exists(temp_output_folder):
+        os.mkdir(temp_output_folder)
+    train_time_in_seconds = config.max_runtime_seconds
+    log_path = os.path.join(temp_output_folder, 'log.txt')
+    sub_command = config.type
+    column_num = dataset.train.X.shape[1]
+    columns=['column_{}'.format(i) for i in range(column_num)]
+    train_df = pd.DataFrame(dataset.train.X, columns=columns)
+    train_df['label'] = dataset.train.y
+    test_df = pd.DataFrame(dataset.test.X, columns=columns)
+    test_df['label'] = dataset.test.y
+    train_dataset = os.path.join(temp_output_folder, 'train.csv')
+    test_dataset = os.path.join(temp_output_folder, 'test.csv')
+    log.info("saving train to {}".format(train_dataset))
+    train_df.to_csv(train_dataset, index=False, header=True)
+    test_df.to_csv(test_dataset, index=False, header=True)
```

Comment:
does MLNET support ARFF files?
If so you can access the files path directly using `dataset.train.path`, as well as training information like `dataset.target`, `dataset.predictors`... 

Those work not only with OpenML but also when using a custom file (which could be a csv...).
So it's much preferable to pass the files directly to MLNet and avoid all those data conversion that can be very expensive for larger datasets.
If MLNet doesn't support `arff` files, but only `csv`, I would suggest to do the conversion only for those.

Reply:
Unfortunately, MLNet.cli only support csv currently, so I guess the only thing I can do here is to avoid conversion when dataset's format is csv? Could you point out the code/property I can use to determine the file format?

Reply:
you could check the extension of `dataset.train.path` for example